AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/RecyclerViewMrecAdActivity.kt

```diff
@@ -0,0 +1,160 @@
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.applovin.enterprise.apps.demoapp.R
+import com.applovin.mediation.MaxAd
+import com.applovin.mediation.MaxAdFormat
+import com.applovin.mediation.MaxAdViewAdListener
+import com.applovin.mediation.MaxError
+import com.applovin.mediation.ads.MaxAdView
+import com.applovin.sdk.AppLovinSdkUtils
+
+class RecyclerViewMrecAdActivity : AppCompatActivity(), MaxAdViewAdListener {
+    private val AD_VIEW_COUNT = 5
+    private val AD_INTERVAL = 10
+    private val sampleData = ArrayList("ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".chunked(1))
+    private val adViews = ArrayList<MaxAdView>(AD_VIEW_COUNT)
+    private lateinit var adapter: CustomRecyclerAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_mrec_recycler_view)
+        setTitle(R.string.activity_recycler_view_mrecs)
+
+        adapter = CustomRecyclerAdapter(this, sampleData)
+
+        val recyclerView = findViewById<RecyclerView>(R.id.mrec_recycler_view)
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.adapter = adapter
+
+        configureAdViews()
+    }
+
+    private fun configureAdViews() {
+        // Insert rows at each interval to be used to display an ad
+        for (i in sampleData.indices step AD_INTERVAL) {
+            sampleData.add(i, "")
+            adapter.notifyItemInserted(i)
+        }
+
+        for (i in 0 until AD_VIEW_COUNT) {
+            val adView = MaxAdView("YOUR_AD_UNIT_ID", MaxAdFormat.MREC, this)
+            adView.setListener(this)
+
+            // Set this extra parameter to work around SDK bug that ignores calls to stopAutoRefresh()
+            adView.setExtraParameter("allow_pause_auto_refresh_immediately", "true")
+            adView.stopAutoRefresh()
+
+            // Load the ad
+            adView.loadAd()
+            adViews.add(adView)
+        }
+    }
+
+    // RecyclerAdapter and ViewHolder
+
+    private enum class ViewHolderType {
+        AD_VIEW,
+        CUSTOM_VIEW
+    }
+
+    inner class CustomRecyclerAdapter(private val activity: Activity, val data: List<String>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+            return when (ViewHolderType.values()[viewType]) {
+                ViewHolderType.AD_VIEW -> AdViewHolder(activity.layoutInflater.inflate(R.layout.activity_mrec_ad_view_holder, parent, false))
+                ViewHolderType.CUSTOM_VIEW -> CustomViewHolder(activity.layoutInflater.inflate(R.layout.activity_mrec_custom_view_holder, parent, false))
```

Comment:
Since this custom view holder is functionally the same as the existing `activity_native_recycler_view_holder.xml` (a single text view), we might as well reuse that XML here.

Let's rename the `activity_native_recycler_view_holder.xml` to make it more generic, i.e., `activity_text_recycler_view_holder` and do the same for Java.